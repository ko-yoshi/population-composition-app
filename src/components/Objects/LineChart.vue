<template>
  <template v-if="chartData">
    <Line class="contents" :options="options" :data="chartData" />
  </template>
</template>

<script lang="ts">
  import { Line } from 'vue-chartjs'
  import {
    Chart as ChartJS,
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend,
  } from 'chart.js'

  ChartJS.register(Title, Tooltip, Legend, PointElement, LineElement, CategoryScale, LinearScale)
</script>
<script lang="ts" setup>
  const { getLineChartData, getLineChartOptions } = useLineChartStore()

  const chartData = computed(() => {
    return getLineChartData()
  })

  const options = computed(() => {
    return getLineChartOptions()
  })
</script>
<style lang="scss" scoped>
  .contents {
    height: 55vh;

    @media screen and (max-width: 480px) {
      height: 50vh;
    }
  }
</style>
