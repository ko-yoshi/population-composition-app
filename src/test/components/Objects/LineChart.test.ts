import { shallowMount } from '@vue/test-utils'
import { expect, test, describe, vi } from 'vitest'
import LineChart from '~/components/Objects/LineChart.vue'

vi.mock('#app', () => ({}))
vi.mock('~/composables/useLineChartStore', () => ({
  useLineChartStore: vi.fn(() => ({
    state: {
      value: { apiResults: [{ 0: { label: '総人口', data: [{ year: 1980, value: 12345 }] } }] },
    },
    getLineChartData: vi.fn(() => ({
      labels: ['1月', '2月', '3月', '4月', '5月', '6月'],
      datasets: [
        {
          label: 'データセット1',
          data: [10, 20, 30, 40, 50, 60],
        },
      ],
    })),
    getLineChartOptions: vi.fn(),
  })),
}))

describe('LineChart.vue', () => {
  const wrapper = shallowMount(LineChart)

  test('LineChartがレンダリングされる', () => {
    expect(wrapper.findComponent(LineChart).exists()).toBe(true)
  })

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
