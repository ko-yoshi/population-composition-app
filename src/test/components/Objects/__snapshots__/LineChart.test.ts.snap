// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LineChart.vue > snapshot 1`] = `"<anonymous-stub data-v-11f98ee5=\\"\\" data=\\"[object Object]\\" options=\\"[object Object]\\" plugins=\\"\\" datasetidkey=\\"label\\" class=\\"contents\\"></anonymous-stub>"`;
